mqtt/mqtt-bridge/src/bridge.rs

```diff
@@ -0,0 +1,41 @@
+use crate::settings::Settings;
+use anyhow::Result;
+use tracing::info;
+
+pub struct NestedBridge {
+    settings: Settings,
+}
+
+impl NestedBridge {
+    pub fn new(settings: Settings) -> Self {
+        NestedBridge { settings }
+    }
+
+    pub async fn start(&self) {
+        info!("Starting nested bridge...{:?}", self.settings);
+
+        self.connect_to_local().await;
+        self.connect_upstream().await;
+    }
+
+    async fn connect_upstream(&self) {
+        info!("getting sas token from edgelet");
+        let token = self.get_sas_token().await;
```

Comment:
I think you need to propagate the error handling up till `start` method.

Reply:
removed the call to get sas token, for now method is just logging, will handle errors when implemented